src/Catalog/Helpers/Utils.cs

```diff
@@ -45,40 +52,131 @@ public static string GetResource(string resName)
+            var rdfxml = SafeXmlTransform(nuspec.CreateReader(), XslTransformNuSpec, arguments);
+
+            var doc = SafeCreateXmlDocument(rdfxml.CreateReader());
+            if (normalizeXml)
             {
-                transform.Transform(nuspec.CreateReader(), arguments, writer);
+                NormalizeXml(doc);
             }
 
             RdfXmlParser rdfXmlParser = new RdfXmlParser();
-            XmlDocument doc = new XmlDocument();
-            doc.Load(rdfxml.CreateReader());
             IGraph graph = new Graph();
             rdfXmlParser.Load(graph, doc);
 
             return graph;
         }
 
-        static XslCompiledTransform CreateTransform(string name)
+        private static void NormalizeXml(XmlNode xmlNode)
+        {
+            if (xmlNode.Attributes != null)
+            {
+                foreach (XmlAttribute attribute in xmlNode.Attributes)
+                {
+                    attribute.Value = attribute.Value.Normalize(NormalizationForm.FormC);
+                }
+            }
+
+            if (xmlNode.Value != null)
+            {
+                xmlNode.Value = xmlNode.Value.Normalize(NormalizationForm.FormC);
+                return;
+            }
+
+            foreach (XmlNode childNode in xmlNode.ChildNodes)
+            {
+                NormalizeXml(childNode);
+            }
+        }
+
+        internal static XmlDocument SafeCreateXmlDocument(XmlReader reader = null)
+        {
+            // CodeAnalystic / XmlDocument: set the resolver to null or instance
+            var xmlDoc = new XmlDocument()
+            {
+                XmlResolver = null
+            };
+
+            if (reader != null)
+            {
+                xmlDoc.Load(reader);
+            }
+
+            return xmlDoc;
+        }
+
+        private static XDocument SafeXmlTransform(XmlReader reader, string xsltResourceName, XsltArgumentList arguments = null)
+        {
+            XDocument result = new XDocument();
+            using (XmlWriter writer = result.CreateWriter())
+            {
+                SafeXmlTransform(reader, writer, xsltResourceName, arguments);
+            }
+            return result;
+        }
+
+        // consider removing overload with dead code (CreateHtmlView)
+        private static void SafeXmlTransform(XmlReader reader, XmlWriter writer, string xsltResourceName, XsltArgumentList arguments = null)
+        {
+            if (arguments == null)
+            {
+                arguments = new XsltArgumentList();
+            }
+
+            var transform = SafeLoadXslTransform(xsltResourceName);
+
+            // CodeAnalysis / XslCompiledTransform.Transform: set resolver property to null or instance
+            transform.Transform(reader, arguments, writer, documentResolver: null);
+        }
+
+        private static XslCompiledTransform SafeLoadXslTransform(string resourceName)
```

Comment:
Did you get the contents of this function from anywhere or was it a change you made for Fortify?